src/main/resources/db/migration/V1_49_0__contract_type.sql

```diff
@@ -0,0 +1,19 @@
+create table contract_type(
+  id uuid not null,
+  name text not null,
+
+  primary key (id)
+);
+
+alter table dynamic_framework_contract
```

Comment:
this script is gonna alter existing contracts in dev / research

can we pull out the service_category_id from each record, and set the new contract type to be a contract type which has a single service category with the same id?

Reply:
i think this should do it 

```
alter table dynamic_framework_contract
    add column contract_type_id uuid,
    add constraint fk__contract_type__contract_type_id foreign key(contract_type_id) references contract_type;
update dynamic_framework_contract
    set contract_type_id = subquery.contract_type_id
    from (select id, t.contract_type_id
      from dynamic_framework_contract c
      inner join contract_type_service_category t
      on t.service_category_id = c.service_category_id) as subquery
    where dynamic_framework_contract.id = subquery.id;
alter table dynamic_framework_contract
    drop column service_category_id,
    alter column contract_type_id set not null;
    ```